src/tools/perf/api/libperf.h

```diff
@@ -130,6 +130,8 @@ typedef struct ucx_perf_result {
 
 typedef void (*ucx_perf_rte_progress_cb_t)(void *arg);
 
+typedef ucs_status_t (*ucx_perf_rte_setup_func_t)(void *arg);
```

Comment:
maybe the setup/cleanup could accept perftest context type directly instead of void*?

Reply:
this would enable typ e checking, but any user of libperf would have to name the structure perftest_context.